File: RabbitMqReceive/WeeklyTaskReportModel.cs
namespace RabbitMqReceive;

public sealed class WeeklyTaskReportModel
{
    public Guid Id { get; set; }
    public DateTime WeekStartDate { get; set; }
    public DateTime WeekEndDate { get; set; }
    public int WeekNumber { get; set; }
    public int Year { get; set; }
    public int TotalTasks { get; set; }
    public int CompletedTasks { get; set; }
    public int PendingTasks { get; set; }
    public double CompletionPercentage { get; set; }
}

File: TaskListAngular/src/app/core/models/weekly-task-report.model.ts
export interface WeeklyTaskReport {
  id: string;
  weekStartDate: Date;
  weekEndDate: Date;
  weekNumber: number;
  year: number;
  totalTasks: number;
  completedTasks: number;
  pendingTasks: number;
  completionPercentage: number;
}
